lib/rust/ensogl/core/src/display/world.rs

```diff
@@ -51,107 +53,195 @@ impl Uniforms {
+    rc: Rc<WorldDataWithLoop>,
 }
 
 impl World {
+    /// Constructor.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Constructor modifier. Displays the default scene in the provided path.
+    pub fn displayed_in(self, dom: impl DomPath) -> Self {
+        self.default_scene.display_in(dom);
+        self
+    }
+
+    /// Keeps the world alive even when all references are dropped. Use only if you want to keep one
+    /// instance of the world forever.
+    pub fn keep_alive_forever(&self) {
+        mem::forget(self.clone_ref())
+    }
+}
+
+impl Deref for World {
+    type Target = WorldDataWithLoop;
+    fn deref(&self) -> &Self::Target {
+        &*self.rc
+    }
+}
+
+impl display::Object for World {
+    fn display_object(&self) -> &display::object::Instance {
+        self.default_scene.display_object()
+    }
+}
+
+impl<'t> From<&'t World> for &'t Scene {
+    fn from(world: &'t World) -> Self {
+        &world.default_scene
+    }
+}
+
+
+
+// =========================
+// === WorldDataWithLoop ===
+// =========================
+
+/// World data with a main loop implementation.
+///
+/// # Main Loop Performance
+/// Any code repeated on each iteration of the Main Loop (each "frame") must be written with a high
+/// care for performance. Any changes that has a chance of negatively impacting the constant
+/// overhead of the main loop needs *explicit* explanation, review, and acceptance *at the design
+/// stage* of the proposed new implementation, from performance perspective, with an
+/// explicit note of the fact of Main Loop impact.
+///
+/// Rationale: the "Main Loop" contains the code comprising a GUI rendering "frame" (term
+/// originating from a "still frame" term in filmmaking). The speed at which the Main Loop executes
+/// directly translates to the perceived performance of the GUI, and the FPS (frames per second)
+/// metric, impacting Users' experience with the application.
+#[derive(Debug)]
+pub struct WorldDataWithLoop {
+    main_loop:        animation::DynamicLoop,
+    main_loop_handle: callback::Handle,
+    data:             WorldData,
+}
+
+impl WorldDataWithLoop {
+    /// Constructor.
+    pub fn new() -> Self {
+        let data = WorldData::new();
+        let main_loop = animation::DynamicLoop::new();
+        let main_loop_handle = main_loop.on_frame(f!([data](t:animation::TimeInfo) {
+            let previous_frame_stats = data.stats.begin_frame();
+            if let Some(stats) = previous_frame_stats {
+                data.on.prev_frame_stats.run_all(&stats);
+            }
+            data.on.before_frame.run_all(t);
+            data.uniforms.time.set(t.local);
+            data.scene_dirty.unset_all();
+            data.default_scene.update(t);
+            data.default_scene.render();
+            data.on.after_frame.run_all(t);
+            data.stats.end_frame();
+        }));
+        Self { main_loop, main_loop_handle, data }
```

Comment:
Make this a separate function, so we could use it in native tests (where request_animation_frame does nothing).